modules/ref_fb_module/src/statistics_fb_impl.cpp

```diff
@@ -19,16 +22,32 @@ StatisticsFbImpl::StatisticsFbImpl(const ContextPtr& ctx, const ComponentPtr& pa
     avgSignal.setDomainSignal(domainSignal);
     rmsSignal.setDomainSignal(domainSignal);
 
-    createAndAddInputPort("input", PacketReadyNotification::Scheduler);
+    if (config.assigned() && config.hasProperty("UseMultiThreadedScheduler"))
+    {
+        if (config.getPropertyValue("UseMultiThreadedScheduler"))
+            packetReadyNotification = PacketReadyNotification::Scheduler;
+        else
+            packetReadyNotification = PacketReadyNotification::SameThread;
+    }
+    else
+    {
+        packetReadyNotification = PacketReadyNotification::Scheduler;
+    }
+
+    createAndAddInputPort("input", packetReadyNotification);
 }
```

Comment:
This can be simplified to 
```
    packetReadyNotification = PacketReadyNotification::Scheduler;
    if (config.assigned() && config.hasProperty("UseMultiThreadedScheduler"))
    {
        if (!config.getPropertyValue("UseMultiThreadedScheduler"))
            packetReadyNotification = PacketReadyNotification::SameThread;
    }
```

Or if you prefer 1 liners:
```
const bool useScheduler = config.assigned() && 
                          config.hasProperty("UseMultiThreadedScheduler") && 
                          config.getPropertyValue("UseMultiThreadedScheduler");
packetReadyNotification = useScheduler ? PacketReadyNotification::Scheduler : PacketReadyNotification::SameThread;
```

Reply:
The first one I agree with, but I think in the 1 liner example is wrong here, because we want to default to Scheduler in case the property is not set.

I think the correct 1 liner would be:
```
const bool useScheduler = config.assigned() && 
                          config.hasProperty("UseMultiThreadedScheduler") && 
                          config.getPropertyValue("UseMultiThreadedScheduler");
packetReadyNotification = !useScheduler ? PacketReadyNotification::SameThread : PacketReadyNotification::Scheduler;
```

However I prefer the version from your first example. I'll update the code accordingly.

Reply:
Or even more concise:
```
packetReadyNotification = PacketReadyNotification::Scheduler;
if (config.assigned() && config.hasProperty("UseMultiThreadedScheduler") && !config.getPropertyValue("UseMultiThreadedScheduler"))
{
    packetReadyNotification = PacketReadyNotification::SameThread;
}
```
I'll update in both trigger and statistics function block.